### PR TITLE
Feature/bvh

### PR DESCRIPTION
**Pull request by andrearastelli**
BVH and AABB classes created.

New way of checking object collision using Bounding Volumes.

test_bvh

### CODE REVIEW

**Comment by andrearastelli** [comment]
Rebased with `develop`

File: src/sphere.h [hunk 1]
```diff
@@ -19,6 +19,7 @@ class Sphere : public Hitable
 	Sphere(Vec3 center, float radius, Material *mat) : center{ center }, radius{ radius }, mat_ptr{ mat } {}
 
     virtual bool hit(const Ray &r, float tmin, float tmax, HitRecord &rec) const;
+	virtual bool bounding_box(float t0, float t1, AABB &box) const;
```
Comment (andrearastelli) [inline src/sphere.h]: - [ ] Spaces instead of tabs.

**Comment by andrearastelli** [comment]
@Ale32 I think you can review this, now... and we can move on with textures 😁

File: src/aabb.h [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        auto invD = 1.0f / r.direction()[a];
+
+        auto t0 = (min()[a] - r.origin()[a]) * invD;
+        auto t1 = (max()[a] - r.origin()[a]) * invD;
+
+        if (invD < 0.0f)
+            std::swap(t0, t1);
+
+        tmin = t0 > tmin ? t0 : tmin;
+        tmax = t1 < tmax ? t1 : tmax;
```
Comment (Ale32) [inline src/aabb.h]: Why not using `ffmax` and `ffmin` here?

**Pull request merged**